tests/PHPStan/Rules/PhpDoc/VarTagChangedExpressionTypeRuleTest.php

```diff
@@ -48,4 +48,9 @@ public function testAssignOfDifferentVariable(): void
 		]);
 	}
 
+	public function testBug10130(): void
+	{
+		$this->analyse([__DIR__ . '/data/bug-10130.php'], []);
```

Comment:
Your expected output is "Report even third widening." but your change silences all.

BTW the reasoning behind the original code is that sometimes you want to do this:

```php
/** @var int|null */
$a = null;
foreach ($list as $v) {
    // sometimes assign $a
}
```

So I'd accept a change that still does not report this (including array shapes etc.):

```php
/** @var int|null */
$a = null;
```

But reports this:

```php
// $null is null
/** @var int|null */
$a = $null;
```

It could be achieved by allowing this only for `ConstFetch`, `Scalar` nodes, and `Array_` node.

Reply:
> Your expected output is "Report even third widening." but your change silences all.

I did not, just the test was wrong. Fixed

> It could be achieved by allowing this only for ConstFetch, Scalar nodes, and Array_ node.

Yeah something like that is needed. Let me try.

Reply:
The only issue was Array_ so added exception for that.